### ISSUE TITLE
bug: missing MAX_HIGH, MAX_LOW constants in OS

### ISSUE THREAD

**Issue by ftheirs** [starkware-libs/cairo#6499]
# Bug Report
Hi everyone,
I've encountered some strange behavior after building `os.cairo`. In the debug_info section, I can see that some functions include the `split_felt` function, but I'm facing an issue when I try to run this function within a transaction.

The error message I'm getting is:
`inner_exc error: Got an exception while executing a hint: Missing constant: MAX_HIGH
`

The constants are defined [here](https://github.com/starkware-libs/cairo-lang/blob/54d7e92a703b3b5a1e07e9389608178129946efc/src/starkware/cairo/common/math.cairo#L119-L149) in math.cairo, However, they cannot be found inside the hint implementation from [math_utils.rs](https://github.com/lambdaclass/cairo-vm/blob/3fb0344ce038b3a68cae897c403d1f561cfe8da7/vm/src/hint_processor/builtin_hint_processor/math_utils.rs#L394-L395)

I'm not entirely sure, but this seems like a compiler optimization that might be omitting the function.
The only reference I can see in the compile output from os.cairo is:
```
"starkware.cairo.common.uint256.split_felt": {
    "destination": "starkware.cairo.common.math.split_felt",
    "type": "alias"
},
"starkware.cairo.common.keccak_utils.keccak_utils.split_felt": {
    "destination": "starkware.cairo.common.math.split_felt",
    "type": "alias"
},
```

I created a simple Cairo program that calls `split_felt`, and it works just fine. Additionally, after inspecting the compilation output from this dummy program, I was able to find the `MAX_HIGH` and `MAX_LOG` constants.
So, when there is an explicit call to the function, it appears on the compilation output as expected.

```
"starkware.cairo.common.math.split_felt.MAX_HIGH": {
    "type": "const",
    "value": 10633823966279327296825105735305134080
},
"starkware.cairo.common.math.split_felt.MAX_LOW": {
    "type": "const",
    "value": 0
},
```

The last test I did was to explicitly include the split_felt function in os.cairo ([here](https://github.com/starkware-libs/cairo-lang/blob/a86e92bfde9c171c0856d7b46580c66e004922f3/src/starkware/starknet/core/os/os.cairo#L14)) and rebuild the program. After this, I was able to find the missing constants in the compilation output and process the block without running into any error.

TL; DR:
I was wondering if there is a missing include on os.cairo code, or if it's a more complex issue is related to an compiler optimization that omits `split_felt` from the compiled output.

**Cairo version:**
`Cairo v0.13.2.1
`

**Current behavior:**
When a transaction calls split_felt function, we get this error:
`
Got an exception while executing a hint: Missing constant: MAX_HIGH
`

**Expected behavior:**
The function should be included on the compilation output and it should work.

**Steps to reproduce:**
On [snos](https://github.com/keep-starknet-strange/snos) repository run the script that builds the OS ( `./scripts/setup-tests.sh`) and inspect `os_latest.json` 
Add split_felt to os.cairo and run the script again to check that this time, the missing constants are there.

**Comment by orizi** [comment]
This is the wrong repo for this question - this is the Cairo1 language - you are talking about the old version of the language (cairo0 or cairo-legacy)
